System tests for Joomla 3.4

javigomez opened the pull request:
This pull request contains all the work from @kshitijSharma2014 Google Summer of Code Project of 2014 mentored by @puneet0191 and me (see https://www.google-melange.com/gsoc/project/details/google/gsoc2014/kshitij_sharma2014/5685265389584384). It:
- fix all previous test to make they work with Joomla 3.4
- add integration tests for Modules and Menu Items 
- add a report tool created by Puneet

This is part of the work done by the Automated Testing Working Group. More info can be found at http://volunteers.joomla.org/working-groups/automated-tests-working-group

javigomez committed:
Remove not needed Codecept.phar

puneet0191 committed:
Archive Weblinks tests

Given that Weblinks is now moved to it's own repository and outside Joomla we are archiving the tests.

New tests will be placed in weblinks repository

kshitijSharma2014 committed:
Update tests for Joomla 3.4

All tests have been reviewed to ensure they all work on Joomla 3.4

puneet0191 committed:
Add system test report tool

puneet0191 commented on tests/system/webdriver/Pages/SiteContent/SiteSingleArticlePage.php:
```diff
@@ -1,4 +1,11 @@
+ * @package     Joomla.Tests
+ * @subpackage  Page
+ *
+ * @copyright   Copyright (C) 2005 - 2014 Open Source Matters, Inc. All rights reserved.
```
We might need to update the copyrights! they are 2014 everywhere

wilsonge replied:
I'm afraid you do :)

puneet0191 commented:
Great work! @kshitijSharma2014  Thanks for the efforts :+1:  special thanks to @javigomez for all the help and being a great co-mentor!

kshitijSharma2014 commented:
thank you @puneet0191  and @javigomez  :smile:

brianteeman commented:
Does this replace any of the existing open pull requests labelled as unit tests? If so can they be closed now<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6057">issues.joomla.org/joomla-cms/6057</a>.</sub>

wilsonge commented:
At a quick glance - no

javigomez commented:
Tests works, I'll fix the copyrights tomorrow. Thanks for the checks.

I will also review the remaining Tests pulls

```
phpunit
PHPUnit 3.7.29 by Sebastian Bergmann.

Configuration read from tests/system/webdriver/tests/phpunit.xml.dist

...............................................	................  63 / 184 ( 34%)
............................................................... 126 / 184 ( 68%)
..........................................................

Time: 2.41 hours, Memory: 17.00Mb

OK (184 tests, 7537 assertions)
```

javigomez committed:
remove not used codeception files

This files were never used, they will be added again in the future in it's right place

javigomez committed:
Fix Copyrights in tests and pages
